server/routes/middleware/Token.js

```diff
@@ -4,52 +4,58 @@ const User = require('../../models/User');
-  if (!token)
-    return res.status(403).send({ auth: false, message: 'No token provided.' });
+  if (req.headers['authorization']) {
+    console.log("Verifying token.");
+    var token = req.body.token || req.query.token || req.headers['x-access-token'] || req.headers['authorization'].split(' ')[1]; // normal headers "Authorization: Bearer 2kj234df0ds2f3n40n"
+    if (!token)
+      return res.status(403).send({ auth: false, message: 'No token provided.' });
 
-  jwt.verify(token, privateKey, function (err, decoded) {
-    if (err) {
-      if (err.name == "TokenExpiredError") {
-        console.log("Deleting token from db.")
-        // delete token from UserSession
-        UserSession.findOneAndRemove({
-          token: token
-        }, (err) => {
-          if (err) {
-            return res.status(500).send({
-              success: false,
-              message: "Error: Server error"
-            });
-          }
-        });
-      }
-      return res.status(401).send({ auth: false, err });
-    }
-    // save to request for use in other routes
-    req.user_id = decoded.user_id;
-    req.role = decoded.role;
-    req.token = token;
-    // Check log in status
-    UserSession.findOne({
-      token: token
-    }, null, (err, session) => {
+    jwt.verify(token, privateKey, function (err, decoded) {
       if (err) {
-        return res.status(500).send({
-          success: false,
-          message: "Error: Server error"
-        });
+        if (err.name == "TokenExpiredError") {
+          console.log("Deleting token from db.")
+          // delete token from UserSession
+          UserSession.findOneAndRemove({
+            token: token
+          }, (err) => {
+            if (err) {
+              return res.status(500).send({
+                success: false,
+                message: "Error: Server error"
+              });
+            }
+          });
+        }
+        return res.status(401).send({ auth: false, err });
       }
-      if (!session) {
-        return res.status(401).send({
-          success: false,
-          message: "Error: Invalid token."
-        });
-      }
-      next();
+      // save to request for use in other routes
+      req.user_id = decoded.user_id;
+      req.role = decoded.role;
+      req.token = token;
+      // Check log in status
+      UserSession.findOne({
+        token: token
+      }, null, (err, session) => {
+        if (err) {
+          return res.status(500).send({
+            success: false,
+            message: "Error: Server error"
+          });
+        }
+        if (!session) {
+          return res.status(401).send({
+            success: false,
+            message: "Error: Invalid token."
+          });
+        }
+        next();
+      });
     });
-  });
+  }
+  else {
+    req.role = '';
```

Comment:
Which if does this correspond to? And is it really necessary? It becomes undefined which is the same as ""

Reply:
Yeah This is to just store the value of req.role, so that it returns nothing. The next() is necessary for sure